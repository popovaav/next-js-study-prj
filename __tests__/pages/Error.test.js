import { render } from '../settings/test-utils';
import '@testing-library/jest-dom/extend-expect';
import Error from '../../pages/_error';
import { NOT_FOUND, INTERNAL_SERVER_ERROR } from '../../src/constants/httpStatusCodes';

describe('Sidebar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(<Error statusCode={NOT_FOUND} />);
  });

  test('Check error message in page', () => {
    expect(wrapper.getByText(/pageNotFound/)).toBeInTheDocument();
    wrapper.rerender(<Error statusCode={INTERNAL_SERVER_ERROR} />);
    expect(wrapper.getByText(/internalServerError/)).toBeInTheDocument();
  });

  test('Snapshot checking', () => {
    expect(wrapper.container).toMatchSnapshot();
  });
});
