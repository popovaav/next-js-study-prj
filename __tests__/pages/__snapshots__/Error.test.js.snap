// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Snapshot checking 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <div
        class="errorBody"
      >
        <h1
          class="title"
        >
          pageNotFound
        </h1>
      </div>
    </main>
    <footer
      class="footer"
    >
      <a
        href="https://onix-systems.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        poweredBy
         
        onix
      </a>
    </footer>
  </div>
</div>
`;
